syntax = "proto3";

package metatrader;

service CandleService {

    // Gets an array of the last n candles.
    rpc GetLastCandles (GetLastCandlesReq) returns (CandleList);

    // Gets an array of the last candles since a given date.
    rpc GetCandlesSince (GetCandlesSinceReq) returns (CandleList);
}

enum Timeframe {
  M1 = 0;
  M15 = 1;
  H1 = 2;
  H4 = 3;
  D1 = 4;
}

message Candle {
    int64 time = 1;
    float open = 2;
    float high = 3;
    float low = 4;
    float close = 5;
    float volume = 6;
}

message GetLastCandlesReq {
    string symbol = 1;
    Timeframe timeframe = 2;
    int64 amount = 3;
}

message GetCandlesSinceReq {
    string symbol = 1;
    Timeframe timeframe = 2;
    int64 since = 3;
}

message CandleList {
    repeated Candle candles = 1;
}
